extensions/interactions/LogicProof/LogicProof.js

```diff
@@ -109,33 +130,47 @@ oppia.directive('oppiaInteractiveLogicProof', [
             // We update the message only if the user has added or removed a
             // line break, so that it remains while they work on a single line.
             if (change.text.length > 1 || change.removed.length > 1) {
-              $scope.refreshMessages(editor);
+              $scope.checkForBasicErrors();
             }
           });
+
+          $scope.editor = editor;
         };
 
-        $scope.refreshMessages = function(editor) {
-          if($scope.mistakeMark) {
-            $scope.mistakeMark.clear();
+        $scope.checkForBasicErrors = function() {
+          if (!$scope.messageIsSticky) {
+            $scope.clearMessage();
           }
           try {
             logicProofStudent.validateProof(
               $scope.proofString, $scope.questionInstance);
-            $scope.proofError = '';
           } catch(err) {
-            $scope.proofError = $scope.displayMessage(err.message, err.line);
-            $scope.mistakeMark = editor.doc.markText(
-              {line: err.line, ch: 0},
-              {line: err.line, ch: 100},
-              {className: 'logic-proof-erroneous-line'});
+            $scope.clearMessage();
+            $scope.showMessage(err.message, err.line);
+            $scope.messageIsSticky = false;
           }
           // NOTE: this line is necessary to force angular to refresh the
-          // displayed proofError.
+          // displayed mistakeMessage.
           $scope.$apply();
         };
 
-        $scope.displayMessage = function(message, lineNumber) {
-          return 'line ' + (lineNumber + 1) + ': ' + message;
+        $scope.clearMessage = function() {
+          if ($scope.mistakeMark) {
+            $scope.mistakeMark.clear();
+          }
+          $scope.mistakeMessage = '';
+        }
+
+        $scope.showMessage = function(message, lineNum) {
+          $scope.mistakeMessage = $scope.renderMessage(message, lineNum);
+          $scope.mistakeMark = $scope.editor.doc.markText(
+            {line: lineNum, ch: 0},
+            {line: lineNum, ch: 100},
+            {className: 'logic-proof-erroneous-line'});
+        };
+
+        $scope.renderMessage = function(message, lineNum) {
```

Comment:
Per offline discussion: $scope.constructMessage, perhaps? "Render", like "display", still has connotations of representing/depicting artistically.

In addition, the methods here are named with respect to 'message', but I'm wondering if it may be better to call these things 'mistakeMessage' or 'errorMessage' instead, and use that term consistently throughout this file and the HTML file. Currently I believe we have 'message' and 'mistakeMessage'; in addition, submitProof() makes reference to displayed_message and error_message. In other words, it's starting to feel like we have a lot of different kinds of messages, which I believe is not the case. It would be nice to disambiguate.